view-user/src/components/common/ReachIcon/ReachIcon.tsx

```diff
@@ -1,19 +1,21 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import styles from "./ReachIcon.module.css";
 import classNames from "classnames";
 import Image from "next/image";
 
 interface ReachIconProps {
   onClick: () => void;
+  isOpen: boolean;
+  setIsReachModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
 const ReachIcon = (props: ReachIconProps) => {
   const [colorInversion, setColorInversion] = useState<boolean>(false);
   const handleClick = () => {
     setColorInversion(!colorInversion);
-    props.onClick();
-    //TODO 後でモーダルの開閉を追加する
+    props.setIsReachModalOpen(!props.isOpen);
   };
+
```

Comment:
モーダルが閉じられると、スタイルが元に戻るようになると思います。
```suggestion
  useEffect(() => {
    if (!props.isOpen) {
      setColorInversion(false);
    }
  }, [props.isOpen]);
```